### ISSUE TITLE
List of embedding losses

### ISSUE THREAD

**Issue by hbredin** [pyannote/pyannote-audio#78]
- [ ] https://arxiv.org/pdf/1712.05015.pdf
- [ ] [Exponential Discriminative Metric Embedding in Deep Learning](https://arxiv.org/abs/1803.02504)
- [ ] [A Discriminative Feature Learning Approach for Deep Face Recognition](https://ydwen.github.io/papers/WenECCV16.pdf)
- [ ] [Ring loss: Convex Feature Normalization for Face Recognition](https://arxiv.org/abs/1803.00130)
- [ ] [Spoken Language Identification Using LSTM-Based Angular Proximity](https://pdfs.semanticscholar.org/ab9f/94965f2f41469019b8b70bb07be13d17bf01.pdf)
- [ ] [Focal Loss for Dense Object Detection](https://arxiv.org/abs/1708.02002)
- [ ] [Magnet Loss: Metric Learning with Adaptive Density Discrimination](https://research.fb.com/wp-content/uploads/2016/05/metric-learning-with-adaptive-density-discrimination.pdf?)
- [ ] [MTGAN: Speaker Verification through Multitasking Triplet
Generative Adversarial Networks](https://arxiv.org/pdf/1803.09059.pdf)
- [ ] [Sampling Matters in Deep Embedding Learning](https://arxiv.org/pdf/1706.07567.pdf)
- [ ] [Angular Softmax Loss for End-to-end Speaker Verification](https://arxiv.org/pdf/1806.03464.pdf)
- [ ] [Angular Softmax for Short-Duration Text-independent Speaker Verification](https://www.isca-speech.org/archive/Interspeech_2018/pdfs/1545.pdf)
- [ ] [Robust and Discriminative Speaker Embedding via Intra-Class Distance Variance Regularization](http://publications.idiap.ch/index.php/publications/show/3863)
- [ ] [NormFace: L2 Hypersphere Embedding for Face Verification](http://www.cs.jhu.edu/~alanlab/Pubs17/wang2017normface.pdf)
- [ ] [Improved Large-margin Softmax Loss for Speaker Diarisation](https://arxiv.org/pdf/1911.03970.pdf)
- [ ] [Powerful Speaker Embedding Training Framework by Adversarially Disentangled Identity Representation](https://arxiv.org/abs/1912.02522)
- [ ] [Divide and Conquer the Embedding Space for Metric Learning](https://arxiv.org/abs/1906.05990)

Code 

- https://github.com/cvqluu/Angular-Penalty-Softmax-Losses-Pytorch
  Angular penalty loss functions in Pytorch (ArcFace, SphereFace, Additive Margin, CosFace)

**Comment by hbredin** [comment]
A bunch of losses were added by @juanmc2005.
WIP results are available in this paper: https://arxiv.org/abs/1911.01255
